src/data/polynomial/laurent.lean

```diff
@@ -356,6 +360,111 @@ begin
     simpa using hn }
 end
 
+section support
+
+lemma support_C_mul_T (a : R) (n : ℤ) : (C a * T n).support ⊆ {n} :=
+by simpa only [← single_eq_C_mul_T] using support_single_subset
+
+lemma support_C_mul_T_ne_zero {a : R} (a0 : a ≠ 0) (n : ℤ) : (C a * T n).support = {n} :=
+begin
+  rw ← single_eq_C_mul_T,
+  exact support_single_ne_zero _ a0,
+end
+
+@[simp] lemma to_laurent_support (f : R[X]) :
+  f.to_laurent.support = f.support.map nat.cast_embedding :=
+begin
+  generalize' hd : f.support = s,
+  revert f,
+  refine finset.induction_on s _ _; clear s,
+  { simp only [polynomial.support_eq_empty, map_zero, finsupp.support_zero, eq_self_iff_true,
+      implies_true_iff, finset.map_empty] {contextual := tt} },
+  { intros a s as hf f fs,
+    have : (erase a f).to_laurent.support = s.map nat.cast_embedding := hf (f.erase a) (by simp only
+      [fs, finset.erase_eq_of_not_mem as, polynomial.support_erase, finset.erase_insert_eq_erase]),
+    rw [← monomial_add_erase f a, finset.map_insert, ← this, map_add,
+      polynomial.to_laurent_C_mul_T, support_add_eq, finset.insert_eq],
+    { congr,
+      exact support_C_mul_T_ne_zero (polynomial.mem_support_iff.mp (by simp [fs])) _ },
+    { rw this,
+      exact disjoint.mono_left (support_C_mul_T _ _) (by simpa) } }
+end
+
+end support
+
+section degrees
+
+/--  The degree of a Laurent polynomial takes values in `with_bot ℤ`.
+If `f : R[T;T⁻¹]` is a Laurent polynomial, then `f.degree` is the maximum of its support of `f`,
+or `⊥`, if `f = 0`. -/
+def degree (f : R[T;T⁻¹]) : with_bot ℤ := f.support.max
+
+@[simp] lemma degree_zero : degree (0 : R[T;T⁻¹]) = ⊥ := rfl
+
+@[simp] lemma degree_eq_bot_iff {f : R[T;T⁻¹]} : f.degree = ⊥ ↔ f = 0 :=
+begin
+  refine ⟨λ h, _, λ h, by rw [h, degree_zero]⟩,
+  rw [degree, finset.max_eq_sup_with_bot] at h,
+  ext n,
+  refine not_not.mp (λ f0, _),
+  simp_rw [finset.sup_eq_bot_iff, finsupp.mem_support_iff, ne.def, with_bot.coe_ne_bot] at h,
+  exact h n f0,
+end
+
+section exact_degrees
+
+open_locale classical
+
+lemma degree_C_mul_T (n : ℤ) (a : R) (a0 : a ≠ 0) : (C a * T n).degree = n :=
+begin
+  rw degree,
+  convert finset.max_singleton,
+  refine support_eq_singleton.mpr _,
+  simp only [← single_eq_C_mul_T, single_eq_same, a0, ne.def, not_false_iff, eq_self_iff_true,
+    and_self],
+end
+
+lemma degree_C_mul_T_ite (n : ℤ) (a : R) : (C a * T n).degree = ite (a = 0) ⊥ n :=
+by split_ifs with h h;
+  simp only [h, map_zero, zero_mul, degree_zero, degree_C_mul_T, ne.def, not_false_iff]
+
+lemma degree_T [nontrivial R] (n : ℤ) : (T n : R[T;T⁻¹]).degree = n :=
```

Comment:
@riccardobrasca, I have been avoiding using `T_pow`, since there is technically no power appearing and there is no `T`-without-power.  I can change this (and the remaining lemmas where I used `T` alone), but I thought that this was a little different from `X` of polynomials.

I will not change this, but if you have any objections at all, I will!

Reply:
It should be a `simp` lemma anyway.

Reply:
Ooops, I have now!  :upside_down_face: